src/pages/workspaces/workspace/Dashboard.js

```diff
@@ -270,6 +267,50 @@ const WorkspaceDashboard = _.flow(
     refresh()
   })
 
+  const getCloudInformation = () => {
+    return !googleProject && !azureContext ? [] : [
+      dl({},
+        !!googleProject ? [
+          h(InfoRow, { title: 'Cloud Name' }, [
+            h(GcpLogo, { title: 'Google Cloud Platform', role: 'img', style: { height: 16 } })
+          ]),
+          h(InfoRow, { title: 'Location' }, [bucketLocation ? h(Fragment, [
+            h(TooltipCell, [flag, ' ', regionDescription])
+          ]) : 'Loading...']),
+          h(InfoRow, { title: 'Google Project ID' }, [
+            h(TooltipCell, [googleProject]),
+            h(ClipboardButton, { 'aria-label': 'Copy google project id to clipboard', text: googleProject, style: { marginLeft: '0.25rem' } })
+          ]),
+          h(InfoRow, { title: 'Bucket Name' }, [
+            h(TooltipCell, [bucketName]),
+            h(ClipboardButton, { 'aria-label': 'Copy bucket name to clipboard', text: bucketName, style: { marginLeft: '0.25rem' } })
+          ]),
+          Utils.canWrite(accessLevel) && h(InfoRow, {
+            title: 'Estimated Storage Cost',
+            subtitle: !!storageCost ? `Updated on ${new Date(storageCost.lastUpdated).toLocaleDateString()}` : 'Loading last updated...'
+          }, [storageCost?.estimate || '$ ...']),
+          Utils.canWrite(accessLevel) && h(InfoRow, {
+            title: 'Bucket Size',
+            subtitle: !!bucketSize ? `Updated on ${new Date(bucketSize.lastUpdated).toLocaleDateString()}` : 'Loading last updated...'
+          }, [bucketSize?.usage])
+        ] : [
+          h(InfoRow, { title: 'Cloud Name' }, [
+            h(AzureLogo, { title: 'Microsoft Azure', role: 'img', style: { height: 16 } })
+          ]),
+          h(InfoRow, { title: 'Resource Group ID' }, [
+            h(TooltipCell, [azureContext.managedResourceGroupId]),
+            h(ClipboardButton, { 'aria-label': 'Copy resource group id to clipboard', text: azureContext.managedResourceGroupId, style: { marginLeft: '0.25rem' } })
+          ])
+        ]
+      ),
+      !!googleProject && div({ style: { paddingBottom: '0.5rem' } }, [h(Link, {
+        style: { margin: '1rem 0.5rem' },
+        ...Utils.newTabLinkProps,
+        href: bucketBrowserUrl(bucketName)
+      }, ['Open bucket in browser', icon('pop-out', { size: 12, style: { marginLeft: '0.25rem' } })]
+      )])
+    ]
+  }
```

Comment:
I spent half an hour trying to get the indenting and syntax correct, and I'm giving up. I really don't feel that it makes the code easier to understand.

Reply:
Huh - I thought it'd work as I copied. Thanks for trying and sorry for the struggle.

Reply:
I tried rearranging the code on my own, so I probably just screwed it up. But I think yours wasn't quite right (bucket location at least is duplicated). I don't know… I could never get the syntax exactly correct, nor could I get the indenting so that it would actually collapse nicely.